pkg/controller/cdapmaster/cdapmaster_controller.go

```diff
@@ -41,8 +60,418 @@ func Add(mgr manager.Manager) error {
+	resources = append(resources, *rinfo)
+	return resources, nil
+}
+
+// Adds a resource.Item of ConfigMap type to the given resource.Bag
+func addConfigMapItem(r *alpha1.CDAPMaster, name string, labels map[string]string, templates []string, resources []reconciler.Object) ([]reconciler.Object, error) {
+	// Creates the configMap object
+	configMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: r.Namespace,
+			Labels:    labels,
+		},
+		Data: make(map[string]string),
+	}
+
+	ngdata := &templateBaseValue{
+		Master: r,
+	}
+
+	// Load template files that goes into config map
+	for _, tmplFile := range templates {
+		content, err := logbackFromFile(tmplFile, ngdata)
+		if err != nil {
+			return nil, err
+		}
+		configMap.Data[tmplFile] = content
+	}
+
+	resources = append(resources, reconciler.Object{
+		Type:      k8s.Type,
+		Lifecycle: reconciler.LifecycleManaged,
+		Obj: &k8s.Object{
+			Obj:     configMap.DeepCopyObject().(metav1.Object),
+			ObjList: &corev1.ConfigMapList{},
+		},
+	})
+
+	return resources, nil
+}
+
+// Sets resources to the given object. It uses reflection to find and set the
+// field `Spec.Template.Spec.Containers[0].Resources`
+func setResources(obj interface{}, resources *corev1.ResourceRequirements) {
+	if resources == nil {
+		return
+	}
+	value := reflect.ValueOf(obj).Elem()
+
+	for _, fieldName := range []string{"Spec", "Template", "Spec", "Containers"} {
+		value = value.FieldByName(fieldName)
+		if !value.IsValid() {
+			return
+		}
+	}
+	resourcesValue := value.Index(0).FieldByName("Resources")
+	resourcesValue.Set(reflect.ValueOf(*resources))
+}
+
+// itemFromReader reads the logback xml with template substitution
+func logbackFromFile(t string, data interface{}) (string, error) {
+	tmpl, err := template.New(t).ParseFiles(templateDir + t)
+	if err != nil {
+		return "", err
+	}
+
+	var output strings.Builder
+	err = tmpl.Execute(&output, data)
+	if err != nil {
+		return "", err
+	}
+	return output.String(), nil
+}
+
+// ------------------------------ Handlers -------------------------------------
+
+// Objects - handler Objects
+func (b *Base) Objects(rsrc interface{}, rsrclabels map[string]string, observed, dependent, aggregated []reconciler.Object) ([]reconciler.Object, error) {
+	var resources []reconciler.Object
+	master := rsrc.(*alpha1.CDAPMaster)
+
+	labels := make(reconciler.KVMap)
+	labels.Merge(master.Labels, rsrclabels)
+
+	// Add the cdap and hadoop ConfigMap
+	configs := map[string][]string{
+		"cconf": {"cdap-site.xml", "logback.xml", "logback-container.xml"},
+		"hconf": {"core-site.xml"},
+	}
+
+	for k, v := range configs {
+		_, err := addConfigMapItem(master, getConfigName(master, k), labels, v, resources)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return resources, nil
+}
+
+// Observables for base
+func (b *Base) Observables(rsrc interface{}, labels map[string]string) []reconciler.Observable {
+	return []reconciler.Observable{}
```

Comment:
is this correct? In airflow operator, I see things like 	return k8s.NewObservables().
		WithLabels(labels).
		For(&appsv1.StatefulSetList{}).
		For(&corev1.ServiceList{}).
		Get()

and 	return k8s.NewObservables().
		WithLabels(labels).
		For(&corev1.ConfigMapList{}).
		Get() seems to work here